activerecord/test/cases/validations/uniqueness_validation_test.rb

```diff
@@ -522,6 +522,21 @@ def test_validate_uniqueness_with_non_callable_conditions_is_not_supported
     }
   end
 
+  def test_validate_uniqueness_with_conditions_with_record_arg
+    Topic.validates_uniqueness_of :title, conditions: ->(record) {
+      where(written_on: record.written_on.beginning_of_day..record.written_on.end_of_day)
+    }
+
+    today_midday = Time.current.midday
+
+    todays_topic = Topic.create!(title: "Highlights of the Day", written_on: today_midday)
```

Comment:
RuboCop is complaining because this variable is assigned but unused.

If you want to keep it as documentation, you'll need to add a leading underscore:

```suggestion
    _todays_topic = Topic.create!(title: "Highlights of the Day", written_on: today_midday)
```